### ISSUE TITLE
Return profile and process ID when creating new session

### ISSUE THREAD

**Issue by EvaldasSourceCode** [mozilla/geckodriver#334]
![selenium geckodriver profile](https://cloud.githubusercontent.com/assets/23399313/20212598/f5f90a66-a7fb-11e6-8eac-815d6defe7ca.PNG)

This simple demonstration illustrates issue that geckodriver do not returns to selenium correct profile directory path. The path returned by selenium FirefoxProfile.ProfileDirectory property "C:\Users\evaldas.*******\AppData\Local\Temp\anonymous.f9de0c8592074138a54202a7c111a3c3.webdriver-profile" actually do not exists.

C# source code used to produce this issue:
```
class Program
{
        static void Main(string[] args)
        {
            FirefoxProfile firefoxProfile = new FirefoxProfile(@"C:\Users\evaldas.*******\AppData\Roaming\Mozilla\Firefox\Profiles\u0obl5t9.Selenium");

            IWebDriver firefoxDriver = new FirefoxDriver(firefoxProfile);

            Console.WriteLine("firefoxProfile.ProfileDirectory: " + firefoxProfile.ProfileDirectory);

            firefoxDriver.Navigate().GoToUrl("http://docs.seleniumhq.org/");

            firefoxDriver.Quit();
        }
}
```
The latest selenium version 3.0.0.0 and geckodriver 0.11.1

Let me know if you need more information.

**Comment by EvaldasSourceCode** [comment]
Hello @AutomatedTester 

Could you provide rough estimation when this issue could be resolved? Of course I am in no position to expect to be, but I just wanted to let you know that this issue prevents me to complete my project.

Thanks!

**Comment by andreastt** [comment]
The WebDriver spec doesn’t say we should return a capability with the profile directory, although I do see the use case in returning that and the parent process PID. Perhaps we can expose these as extension capabilities:

```js
{
  …
  "moz:pid": <number>,
  "moz:profile": <string>,
  …
}
```

**Comment by EvaldasSourceCode** [comment]
@andreastt thanks for reply. Now I am not very familiar with all processes, but what I expect that when you accessing `firefoxProfile.ProfileDirectory` it will return current "real" location not path which doesn't exits in file system.

**Comment by andreastt** [comment]
Fixed in [bug 1321278](https://bugzilla.mozilla.org/show_bug.cgi?id=1321278). As of Firefox 51, a `moz:profile` capability will be returned.
